libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -271,6 +271,15 @@ class MapboxNavigation(
      */
     val historyRecorder = MapboxHistoryRecorder(navigationOptions, logger)
 
+    /**
+     * Navigation native experimental API. Do not store a link to [Experimental] object, use it with
+     * `mapboxNavigation.experimental` because [MapboxNavigation] is responsible for it's lifecycle.
+     *
+     * Add a separate gradle dependency to use it
+     * "com.mapbox.navigator:mapbox-navigation-native:x.y.z"
+     */
+    val experimental: com.mapbox.navigator.Experimental
```

Comment:
Can we mark it with the `@ExperimentalMapboxNavigationAPI`?

Reply:
we can't. If we add, we will need to mark the whole `MapboxNavigation` with the same annotation, it will affect all customers, even if they do not use experimental

Reply:
Are you sure this is the case? I quickly tried locally by adding `@ExperimentalMapboxNavigationAPI` to one of the `MapboxNavigation` fields, and only using that field raised an error.

Reply:
right, the issue was with
`experimental = navigator.experimental`
replaced it with 
```
@ExperimentalMapboxNavigationAPI
val experimental: com.mapbox.navigator.Experimental
        get() = navigator.experimental
```